contracts/rosca.sol

```diff
@@ -1,266 +1,258 @@
-        address _feeAddress)
-    {
-        roundPeriodInDays = _roundPeriodInDays;
-        if(roundSum < 1) throw; //if everyone is only required to contribute 0 , whats the point of rosca
-        roundSum = _roundSum;
-        if(_numEpochs < 1) throw;
-        numEpochs = _numEpochs;
-        if(_minParticipants < 2) throw;// there should be at least 2 people to make a group
-        minParticipants = _minParticipants;
-        if(_startTime < now) throw;
-        startTime = _startTime;
-        if(_feeInThousandths < 0) throw; //fee must be non-negative
-        feeToService = _feeInThousandths;
-        if(feeAddress != 0) //feeAddress shouldnt be empty(null)
-        feeAddress = _feeAddress;
-
-
-        //register foreman as a member of ROSCA and increment numRounds in a Epoch to 1
-        members[msg.sender] = USER({paid: 0 , contributed: 0, alive: true, allowance: 0});
-        memberAddress.push(msg.sender);
-        numRounds = 1;
-    }
-
-
-
-    function startRound()
-    {
-
-        if(now < startTime + ( currentRound * currentEpoch  * (roundPeriodInDays * 1 days)))
-            throw;
-        else
+  function ROSCA(
+    uint16 roundPeriodInDays_,
+    uint contributionSize_, //i am assuming this is referring to number epochs to run
+    uint16 numEpochs_,
+    uint8 minParticipants_,
+    uint startTime_,
+    uint16 feeInThousandths_,
+    address feeAddress_)
+  {
+    roundPeriodInDays = roundPeriodInDays_;
+    if(contributionSize_ < MIN_CONTRIBUTION_SIZE) throw; //if everyone is only required to contribute 0 , whats the point of rosca
+    contributionSize = contributionSize_;
+
+    if(numEpochs_ < 1) throw;
+    numEpochs = numEpochs_;
+
+    if(minParticipants_ < 2) throw;// there should be at least 2 people to make a group
+    minParticipants = minParticipants_;
+
+    if(startTime_ < (now + 1 days)) throw;
+    startTime = startTime_;
+
+    if(feeInThousandths_ > MAX_FEE_IN_THOUSANDTH) throw; //fee must be less than max fee
+    serviceFeeInThousandths = feeInThousandths_;
+
+    if(feeAddress != 0) //feeAddress shouldnt be empty(null)
+    feeAddress = feeAddress_;
+
+    //set foreman
+    foreman = msg.sender;
+
+    //register foreman as a member of ROSCA and increment numRounds in a Epoch to 1
+    members[msg.sender] = USER({paid: 0 , contributed: 0, alive: true, pendingWithdrawl: 0});
+    membersAddresses.push(msg.sender);
+    numRoundsInEpoch = 1;
+  }
+
+  /** startRound() check if the round has ended, if so, get the winner of the currentRound's pot
+    * and add the amount to the winner's pending withdrawal
+    * if there were no bid during the round, address at index "currentRoundNumber" is chosen,
+    * if the address chosen already won the pot once in currentEpoch, currentRoundNumber+1 is chosen, untill a winner is found
+    *
+    * after the winner had been declared, bidding process reset and currentRound is incremented, if currentRound+1 is equal numRoundsInEpoch,
```

Comment:
It's great that you explain what the function does.
However, we may want to be less specific in the documentation, otherwise it starts duplicating the code and will eventually fall out of sync.